### PR TITLE
nginx CSP etc

### PR DESCRIPTION
**Pull request by fabianfrz**

### CODE REVIEW

File: www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml [hunk 1]
```diff
@@ -512,8 +512,51 @@
         <Required>N</Required>
       </xssprotection>
       <content_type_options type="BooleanField">
-        <Required>N</Required>
+        <Required>Y</Required>
```
Comment (fichtner) [inline www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml]: model bump

File: www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml [hunk 2]
```diff
@@ -484,5 +484,82 @@
       </flag>
     </http_rewrite>
 
+    <security_header type="ArrayField">
+      <description type="TextField">
+        <Required>Y</Required>
```
Comment (fichtner) [inline www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml]: model bump required

Comment (fabianfrz) [reply www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml]: recommended but not required

**Comment by fabianfrz** [comment]
@fichtner why do you approve it? It is an early stage.

File: www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml (hunk 2 above)
Comment (fabianfrz) [reply www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml]: done

File: www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml [hunk 3]
```diff
@@ -484,5 +497,93 @@
...
+      <csp_default_src_mediastream type="BooleanField">
+        <Required>Y</Required>
+      </csp_default_src_mediastream>
+      <csp_default_src_filesystem type="BooleanField">
+        <Required>Y</Required>
+      </csp_default_src_filesystem>
+      <csp_default_src_none type="BooleanField">
+        <Required>Y</Required>
+      </csp_default_src_none>
+    </security_header>
```
Comment (fabianfrz) [inline www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml]: @fichtner an idea how to not copy that from https://github.com/opnsense/plugins/pull/773/files#diff-8fda16d8ce523643673f4b91b34aea3fR557 to here for all of https://github.com/opnsense/plugins/pull/773/files#diff-ca27b344b01813de50f2239c2091dfb3R24 ?

**Comment by fabianfrz** [comment]
@fichtner @mimugmail now the CSP reached a feature level I like ;)

File: www/nginx/Makefile [hunk 4]
```diff
@@ -1,8 +1,7 @@
 PLUGIN_NAME=          nginx
-PLUGIN_VERSION=       0.4
+PLUGIN_VERSION=       1.1
```
Comment (fichtner) [inline www/nginx/Makefile]: this makes no sense...

File: www/nginx/Makefile [hunk 5]
```diff
@@ -1,8 +1,7 @@
...
 PLUGIN_COMMENT=       Nginx HTTP server and reverse proxy
 PLUGIN_DEPENDS=       nginx
 PLUGIN_MAINTAINER=    franz.fabian.94@gmail.com
-PLUGIN_DEVEL=         yes
```
Comment (fichtner) [inline www/nginx/Makefile]: and this too. I haven't reviewed this for release ;)

File: www/nginx/Makefile (hunk 4 above)
Comment (fabianfrz) [reply www/nginx/Makefile]: sure it does 0.4 is 1.0 for 18.1 series. 1.0 should have been released with 18.7.1

Comment (fichtner) [reply www/nginx/Makefile]: nginx was never on 18.1

Comment (fabianfrz) [reply www/nginx/Makefile]: @fichtner then some plans got terribly wrong...

Comment (fichtner) [reply www/nginx/Makefile]: we can release 1.0 in 18.7.2 if you want, but you will have to ask or let me know explicitly. you need to be on top of this ;)

**Pull request merged**